### PR TITLE
Return JSON errors instead of HTML

### PR DESCRIPTION
**Pull request by romanyg**
Created middleware into `app/middleware/defaultError.js` and hooked it up to routes handlers `app/routes/index.js`

### CODE REVIEW

**Review by zuse-z3 — APPROVED** [review]
LGTM!

**Review by jgraeger — CHANGES_REQUESTED** [review]
First of all: Many thanks for your contribution. Right now this is about an university's project and we did not expect somebody to actual contribute to this ☺️ 

To enforce a consistent code style, we use ESLint on this project. I can see that some parts of the code are not compliant with out style guide (e.g. double-quotes strings instead of single-quotes). Please run `ESLint` locally and apply fixes. Then we're good to go :) You can also run `eslint --fix` to automatically fix all the issues.

In case of questions: You can find my email in my github profile :)

File: app/middleware/defaultError.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const notFoundError = (req, res) => {
+  res.status(404).json({ error: 'not found' });
+};
+
+// Express recongizes the error handler by the number of arguments
+// the function accepts, so `next` should be defined even if it is
+// not used inside the handler. ESLint will complain about unused
+// vars, we will disable it by no-unused-vars rule.
+/* eslint-disable no-unused-vars */
```
Comment (jgraeger) [inline app/middleware/defaultError.js:9]: You point out a general issue with eslint and express here, so I think I will move you're explanation to the ESLint config and add a more general rule there :)

**Review by jgraeger — APPROVED** [review]
Thx!  🎉

**Pull request merged**